src/topic_training/src/Begin_Here.md

```diff
@@ -0,0 +1,264 @@
+
+The callback function for a Subscriber looks something like this:
+
+``` C++
+void chatterCallback(const std_msgs::Float64::ConstPtr& msg)
+{
+   pi = msg->data;
+}
+```
+This function will get called when a new message has arrived on the topic `plot`. The message received by the Subscriber, sent by the Publisher, is denoted as `msg`.
+
+Publishing a message on a topic would look something like this:
+
+``` C++
+std_msgs::String msg;
+msg.data = sstream.str();
+pub.publish(msg);
+ros::spinOnce();
+```
+
+To break this down,
+
+``` C++
+std_msgs::String msg;
+```
+Initialize `msg` as the desired type, `std_msgs::String`.
+
+``` C++
+msg.data = sstream.str();
+```
+Set the data of `msg` as what you want the message to send. In this case, `sstream.str()` is a String concatenated with a float value.
+
+``` C++
+pub.publish(msg);
+```
+Publish the message using the previously initialized Publisher `pub`.
+
+``` C++
+ros::spinOnce();
+```
+There is also the function `ros::spin()`. The main difference is that `ros::spinOnce()` is used when other lines of code, and processing arriving messages, need to be executed along side each other, such as in a while loop. Whereas `ros::spin()` should be called at the end of a file.
+
+### Python
+
+A basic setup for ROS topics in Python would look something like this:
+
+``` Python
+rospy.init_node('Demo', anonymous=True)
+rospy.Subscriber('plot', Float64, callback)
+pub = rospy.Publisher('chatter', String, queue_size=QUEUE_SIZE)
+```
+
+To break this down,
+
+``` Python
+rospy.init_node('Demo', anonymous=True)
+```
+Initialize ROS and specify the name of the node (which is the name of the file in this case).
+
+``` Python
+rospy.Subscriber('plot', Float64, callback)
+```
+Initialize the Subscriber. State the name of the topic (`plot`), the message type we will be publishing (float, which is actually in the class std_msgs.msg.Float64), and the function that will be called when a message is received (callBack(data)).
+
+``` Python
+pub = rospy.Publisher('chatter', String, queue_size=QUEUE_SIZE)
+```
+Initialize the Publisher. State the name of the topic (`chatter`), the message type we will be publishing (String, which is actually in the class std_msgs.msg.String), and the queue size (1). The queue size is the maximum number of messages kept as a buffer before throwing away old ones if we are publishing too quickly.
+
+The callback function for a Subscriber looks something like this:
+
+``` Python
+def callback(data: Float64) -> None:
+    global pi 
+    pi = data.data
+```
+This function will get called when a new message has arrived on the topic `plot`. The message received by the Subscriber, sent by the Publisher, is denoted as `msg`.
+
+Publishing a message on a topic would look something like this:
+
+``` Python
+pi_str = "currently at %s" % pi
+pub.publish(hello_str)
```

Comment:
The variable you are publishing here is a separate variable. Not sure if this is intentional or not.

Reply:
Fixed in bcee277.